lms/static/js/demographics_collection/DemographicsCollectionModal.jsx

```diff
@@ -86,42 +98,52 @@ class DemographicsCollectionModal extends React.Component {
     try {
       await this.jwtTokenService.getJwtToken();
       response = await fetch(`${this.props.demographicsBaseUrl}/demographics/api/v1/demographics/${this.props.user}/`, requestOptions);
-      // if the user has not yet bee created in the demographics service, we need to make a post to create an entry
-      if (response.status !== 200) {
-        // we get a 404 if the user resource does not exist in demographics, which is expected.
-        if (response.status === 404) {
-          try {
-            const postUrl = `${this.props.demographicsBaseUrl}/demographics/api/v1/demographics/`;
-            requestOptions.method = 'POST'
-            requestOptions.body = JSON.stringify({
-              user: this.props.user,
-            });
-            const csrfToken = await this.csrfTokenService.getCsrfToken(url);
-            requestOptions.headers['X-CSRFToken'] = csrfToken;
-            Cookies.set('demographics_csrftoken', csrfToken);
-            response = await fetch(postUrl, requestOptions);
-            // A 201 is a created success message. if we don't get a 201, throw an error.
-            if (response.status !== 201) {
-              const error = await response.json();
-              throw error.detail;
-            }
-          } catch (error) {
-            this.setState({ loading: false, error: true, errorMessage: error });
-          }
-        } else {
-          const error = await response.json();
-          throw error.detail;
+      // We found the user, return their data
+      if (response.status === 200) {
+        data = await response.json();
+        if (data[FIELD_NAMES.ETHNICITY]) {
+          // map ethnicity data to match what the UI requires
+          data[FIELD_NAMES.ETHNICITY] = this.reduceEthnicityArray(data[FIELD_NAMES.ETHNICITY]);
         }
+        this.setState({ options: options.actions.POST, loading: false, selected: data });
       }
-
-      data = await response.json();
-      if (data[FIELD_NAMES.ETHNICITY]) {
-        // map ethnicity data to match what the UI requires
-        data[FIELD_NAMES.ETHNICITY] = this.reduceEthnicityArray(data[FIELD_NAMES.ETHNICITY]);
+    } catch (e) {
+      // an error other than "no entry found" occured
+      if (response.status !== 404) {
+        this.setState({ loading: false, error: true, errorMessage: error });
```

Comment:
Overall, this line is duplicated a few times (lines 113, 125, and 145) in this method. Even if it's a one liner, maybe we want something like a `setErrorState` utility method?